sdk/src/tx/txHandler.ts

```diff
@@ -0,0 +1,593 @@
+		}
+
+        [wallet, confirmationOpts] = this.getProps(wallet, confirmationOpts);
+
+		tx.feePayer = wallet.publicKey;
+		recentBlockhash = recentBlockhash ? recentBlockhash : (
+			await this.connection.getLatestBlockhash(confirmationOpts.preflightCommitment)
+		);
+		tx.recentBlockhash = recentBlockhash.blockhash;
+
+        this.addHashAndExpiryToLookup(recentBlockhash);
+
+		const signedTx = await this.signTx(tx, additionalSigners);
+
+		return signedTx;
+	}
+
+    private isVersionedTransaction(tx: Transaction | VersionedTransaction) {
+        return (tx as VersionedTransaction)?.message && true;
+    }
+
+    private getTxSigFromSignedTx(signedTx: Transaction | VersionedTransaction) {
+        if (this.isVersionedTransaction(signedTx)) {
+            return bs58.encode(Buffer.from((signedTx as VersionedTransaction).signatures[0])) as string;
+        } else {
+            return bs58.encode(Buffer.from((signedTx as Transaction).signature)) as string;
+        }
+    }
+    
+    private getBlockhashFromSignedTx(signedTx: Transaction | VersionedTransaction) {
+        if (this.isVersionedTransaction(signedTx)) {
+            return (signedTx as VersionedTransaction).message.recentBlockhash;
+        } else {
+            return (signedTx as Transaction).recentBlockhash;
+        }
+    }
+
+    private async signTx(
+		tx: Transaction,
+        additionalSigners: Array<Signer>,
+        wallet?: Wallet,
+	): Promise<Transaction> {
+
+        [wallet] = this.getProps(wallet);
+
+        additionalSigners
+			.filter((s): s is Signer => s !== undefined)
+			.forEach((kp) => {
+				tx.partialSign(kp);
+			});
+            
+		const signedTx = await wallet.signTransaction(tx);
+
+        // Turn txSig Buffer into base58 string
+        const txSig = this.getTxSigFromSignedTx(signedTx);
+
+        this.handleSignedTxData([
+            {
+                txSig,
+                signedTx,
+                blockHash: this.getBlockhashFromSignedTx(signedTx),
+            }
+        ]);
+
+		return signedTx;
+	}
+
+    public async signVersionedTx(
+		tx: VersionedTransaction,
+        additionalSigners: Array<Signer>,
+        recentBlockHash?: BlockhashWithExpiryBlockHeight,
+        wallet?: Wallet,
+	): Promise<VersionedTransaction> {
+
+        [wallet] = this.getProps(wallet);
+
+        if (recentBlockHash) {
+            tx.message.recentBlockhash = recentBlockHash.blockhash;
+
+            this.addHashAndExpiryToLookup(recentBlockHash);
+        }
+
+        additionalSigners
+            ?.filter((s): s is Signer => s !== undefined)
+            .forEach((kp) => {
+                tx.sign([kp]);
+            });
+            
+        // @ts-ignore :: TODO : They were either busted before or I'e done something wrong. Wonderin crisp/will's input on this. I'm confused about diff between IWallet and Wallet types. Why does Wallet have methods (e.g. `signVersionedTransaction`) which aren't actually present on the wallet at runtime?
```

Comment:
This is something I'm waiting on an answer on before I wanna merge this. I feel like our `Wallet` types are scattered rn and it would be good to coalesce on one